### PR TITLE
Fix major numerical issue in COCOMeanAveragePrecision

### PR DESCRIPTION
**Pull request by LukeWood**
Noticed this bug during numerical testing, now I have sufficient test coverage of this and have fixed the issue.

In short:
```
ground_truths = utils.filter_boxes(...)
```

was overwriting every single other class if a single one was missing.  Now this is fixed

### CODE REVIEW

File: keras_cv/metrics/coco/recall.py [hunk 1]
```diff
@@ -151,6 +156,8 @@ def update_state(self, y_true, y_pred, sample_weight=None):
             dtype=self.compute_dtype,
         )
 
+        y_pred = utils.sort_bounding_boxes(y_pred, axis=bounding_box.XYXY.CONFIDENCE)
```
Comment (bhack) [inline keras_cv/metrics/coco/recall.py:159]: Is this going to fix also https://github.com/keras-team/keras-cv/issues/328 ?

Comment (LukeWood) [reply keras_cv/metrics/coco/recall.py:159]: yep

Comment (bhack) [reply keras_cv/metrics/coco/recall.py:159]: Do we have gained only -0.01  with al these fixes?

Comment (LukeWood) [reply keras_cv/metrics/coco/recall.py:159]: yep

Comment (LukeWood) [reply keras_cv/metrics/coco/recall.py:159]: in extreme cases though it saves us potentially a result that should be 1.0 showing as 0.0

**Pull request merged**